### DISCUSSION TITLE
something is wrong with structs

### DISCUSSION THREAD

**Discussion by keinPlan** [vfsfitvnm/frida-il2cpp-bridge#420] [General]
got a problem with a function (UnityVersion 2021.3.22f1).

if i'm calling the function the parameter seems to be passed in a wrong format.

```
// The Function
System.Void Init(Rk.Rubens.Map.StartMarchCrypt& action);

// STRUCT
struct Rk.Rubens.Map.StartMarchCrypt : System.ValueType
{
    Rk.Rubens.IAtom StartPoint; // 0x10
    Rk.Rubens.IAtom TargetAtom; // 0x18
    System.UInt32 OilCount; // 0x20
    Rk.IList<Rk.Rubens.IAtomHero> TakeHeroes; // 0x28
    System.UInt32 ApproxFinishTime; // 0x30
    System.Void .ctor(Rk.Rubens.IAtom startPoint, Rk.Rubens.IAtom targetAtom, System.UInt32 oilCount, Rk.IList<Rk.Rubens.IAtomHero> takeHeroes, System.UInt32 approxFinishTime); // 0x03682ea0
}
```

i did replace the implementaion and just read the data from the handle passed:

**GAME CALLING:**
   Handle: ->Rk.Rubens.Map.StartMarchCrypt
 HandleType: 0xc2814eeca0
 MemDump 7xuint64:
```
  1967449137824 	// StartPoint
  1968403908576 	// TargetAtom
  1			// OilCount
  1969275494240		// TakeHeroes
  1698436712		// ApproxFinishTime  --- struct Rk.Rubens.Map.StartMarchCrypt END 
  1968609456641	
  8011462212112905538
```
 
 
**ME CALLING:** 
     Handle: ->Rk.Rubens.Map.StartMarchCrypt
 HandleType: 0x1ca52c92180
 MemDump 7xuint64:
```
  1966467562432 	// il2cpp pointer ? 
  0			// il2cpp pointer ? 
  1962936266352 	// StartPoint
  1968403908576 	// TargetAtom
  1 			// OilCount
  1967398819952 	// TakeHeroes
  30 			// ApproxFinishTime --- struct Rk.Rubens.Map.StartMarchCrypt END 
```

also can't read fields of struct the game created... but if reading direct from memory the data are there but without the il2cpp pointers at the start.

currently i just overwritte the il2cpp pointer @ the begining of the struct with the struct data befor passing it to the function ... messy but  seems to work

**Comment by vfsfitvnm** [comment]
Thanks for reporting, are you already using `v0.9.0`? If so, are you able to reproduce with an older version?

**Comment by keinPlan** [comment]
yes i'm using v0.9.0
v0.8.8 same result
v0.7.13 created the correct object

**Reply by vfsfitvnm** [reply]
Thanks, this is indeed a bug. Moved to #421
